ptychonn/position/configs.py

```diff
@@ -0,0 +1,354 @@
+    """
+
+    dp_data_file_path: str = None
+
+    dp_data_file_handle: Any = None
+    """Used as an alternative to `dp_data_file_path`. Should be a `DataFileHandle` object."""
+
+    probe_position_list: Any = None
+    """
+    A ProbePositionList object used for finding nearest neighbors in collective mode.
+    If None, `probe_position_data_path` must be provided.
+    """
+
+    probe_position_data_path: Any = None
+
+    probe_position_data_unit: str = None
+    """Unit of provided probe position. Can be 'nm', 'm', or 'pixel'."""
+
+    pixel_size_nm: float = None
+
+    baseline_position_list: Any = None
+    """Baseline positions. Used by ProbePositionCorrectorChain when the serial mode result is bad."""
+
+    central_crop: Any = None
+    """
+    List or tuple of int. Patch size used for image registration. If smaller than the reconstructed object size,
+    a patch will be cropped from the center.
+    """
+
+    method: str = "collective"
+    """Method for correction. Can be 'serial' or 'collective'"""
+
+    num_neighbors_collective: int = 3
+    """Number of neighbors in collective registration"""
+
+    offset_estimator_order: int = 1
+
+    offset_estimator_beta: float = 0.5
+
+    smooth_constraint_weight: float = 1e-2
+
+    rectangular_grid: bool = False
+
+    stitching_downsampling: int = 1
+
+    random_seed: Any = 123
+
+    debug: bool = False
+
+
+class TrainingConfigDict(ConfigDict):
+    batch_size_per_process: int = 64
+
+    num_epochs: int = 60
+
+    learning_rate_per_process: float = 1e-3
+
+    optimizer: str = "adam"
+    """String of optimizer name or the handle of a subclass of torch.optim.Optimizer"""
+
+    model_save_dir: str = "."
+    """Directory to save trained models"""
+
+    model: Any = None
+    """
+    The model. The three options are:
+    (1) None: the model will be instantiated with the default model class.
+    (2) A object of nn.Module: the model object will be used as provided.
+
+
+    (3) tuple(nn.Module, kwargs): the first element of the tuple is the class handle of a model class, and the
+        second is a dictionary of keyword arguments. The model will be instantiated using these.
+    """
+
+    l1_weight: float = 0
+
+    tv_weight: float = 0
+
+
+class PtychoNNTrainingConfigDict(TrainingConfigDict):
+    height: int = 256
+
+    width: int = 256
+
+    num_lines_for_training: int = 100
+    """Number of lines used for training"""
+
+    num_lines_for_testing: int = 60
+    """Number of lines used for testing"""
+
+    num_lines_for_validation: int = 805
+    """Number of lines used for testing"""
+
+    dataset: Any = None
+    """A torch.Dataset object"""
+
+    validation_ratio: float = 0.003
+    """Ratio of validation set out of the entire dataset"""
+
+    loss_function: Any = None
+    """Can be None (default to L1Loss) or a Callable."""
+
+    dataset_decimation_ratio: float = 1.0
+
+    schedule_learning_rate: bool = True
+
+    pretrained_model_path: str = None
```

Comment:
```suggestion
    pretrained_model_path: typing.Optional[str] = None
```
Any parameter that has `None` as the default value is Optional.

Reply:
Since this is now in the PtychoNN package, modeling training and inference should use the protocols that already exist in the package, so I removed the parameters related to PtychoNN training and inference in f5a3af2.